src/fides/api/api/v1/endpoints/privacy_preference_endpoints.py

```diff
@@ -702,6 +744,9 @@ def save_privacy_preferences(
     Creates historical records for these preferences for record keeping, and also updates current preferences.
     Creates a privacy request to propagate preferences to third party systems if applicable.
     """
+    tc_string: Optional[str] = data.tc_string
+    data = update_request_with_decoded_tc_string_fields(data, db)
```

Comment:
This is the primary change of this PR. A TC string in the request body is decoded if supplied and it's the corresponding sections are added to the request body and then preferences saved as usual.